components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -67,202 +71,188 @@
+        }
 
-            if (StringUtils.isNotEmpty(consumerKey)) {
-                OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
-                try {
-                    String appState = oAuthAppDAO.getConsumerAppState(consumerKey);
-                    if (StringUtils.isEmpty(appState)) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("A valid OAuth client could not be found for client_id: " + consumerKey);
-                        }
-                        OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
-                                .SC_UNAUTHORIZED)
-                                .setError(OAuth2ErrorCodes.INVALID_CLIENT)
-                                .setErrorDescription("A valid OAuth client could not be found for client_id: " +
-                                        consumerKey).buildJSONMessage();
-                        return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                                .build();
-                    }
+        throw new TokenEndpointBadRequestException(null, null, e);
+    }
 
-                    if (!OAuthConstants.OauthAppStates.APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("Oauth App is not in active state.");
-                        }
-                        OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
-                                .SC_UNAUTHORIZED)
-                                .setError(OAuth2ErrorCodes.INVALID_CLIENT)
-                                .setErrorDescription("Oauth application is not in active state.")
-                                .buildJSONMessage();
-                        return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                                .build();
-                    }
-                } catch (IdentityOAuthAdminException e) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Error in getting oauth app state.", e);
-                    }
-                    OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse.SC_NOT_FOUND)
-                            .setError(OAuth2ErrorCodes.SERVER_ERROR)
-                            .setErrorDescription("Error in getting oauth app state.").buildJSONMessage();
-                    return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                            .build();
+    private void validateRepeatedParams(HttpServletRequest request, MultivaluedMap<String, String> paramMap)
+            throws TokenEndpointBadRequestException {
+
+        if (!validateParams(request, paramMap)) {
+            throw new TokenEndpointBadRequestException("Invalid request with repeated parameters.");
```

Comment:
Will there be a way to print the repeated param map in the exception itself

Reply:
log the repeated parameters